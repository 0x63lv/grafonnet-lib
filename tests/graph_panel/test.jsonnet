local grafana = import "grafonnet/grafana.libsonnet";
local graphPanel = grafana.graphPanel;

{
    basic: graphPanel.new("test"),
    advanced: graphPanel.new(
        "test",
        id=2,
        span=8,
        min_span=9,
        format="ms",
        min=10,
        max=10,
        lines=false,
        points=true,
        bars=true,
        dashes=true,
        stack=true,
        repeat="$device",
    ),
    targets: graphPanel.new("with targets")
             + graphPanel.addTarget("foo")
             + graphPanel.addTarget("foo"),
}
