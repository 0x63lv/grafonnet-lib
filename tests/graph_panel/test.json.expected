{
   "advanced": {
      "aliasColors": { },
      "bars": true,
      "dashLength": 10,
      "dashes": true,
      "datasource": null,
      "decimals": 2,
      "fill": 2,
      "id": 2,
      "legend": {
         "avg": false,
         "current": false,
         "max": false,
         "min": false,
         "show": true,
         "total": false,
         "values": false
      },
      "lines": false,
      "linewidth": 2,
      "minSpan": 9,
      "nullPointMode": "null",
      "percentage": false,
      "pointradius": 5,
      "points": true,
      "renderer": "flot",
      "repeat": "$device",
      "seriesOverrides": [ ],
      "spaceLength": 10,
      "span": 8,
      "stack": true,
      "steppedLine": false,
      "targets": [ ],
      "thresholds": [ ],
      "timeFrom": null,
      "timeShift": null,
      "title": "test",
      "tooltip": {
         "shared": true,
         "sort": 0,
         "value_type": "individual"
      },
      "type": "graph",
      "xaxis": {
         "buckets": null,
         "mode": "time",
         "name": null,
         "show": true,
         "values": [ ]
      },
      "yaxes": [
         {
            "format": "ms",
            "label": null,
            "logBase": 1,
            "max": 10,
            "min": 10,
            "show": true
         },
         {
            "format": "ms",
            "label": null,
            "logBase": 1,
            "max": 10,
            "min": 10,
            "show": true
         }
      ]
   },
   "basic": {
      "aliasColors": { },
      "bars": false,
      "dashLength": 10,
      "dashes": false,
      "datasource": null,
      "fill": 1,
      "legend": {
         "avg": false,
         "current": false,
         "max": false,
         "min": false,
         "show": true,
         "total": false,
         "values": false
      },
      "lines": true,
      "linewidth": 1,
      "nullPointMode": "null",
      "percentage": false,
      "pointradius": 5,
      "points": false,
      "renderer": "flot",
      "repeat": null,
      "seriesOverrides": [ ],
      "spaceLength": 10,
      "span": 12,
      "stack": false,
      "steppedLine": false,
      "targets": [ ],
      "thresholds": [ ],
      "timeFrom": null,
      "timeShift": null,
      "title": "test",
      "tooltip": {
         "shared": true,
         "sort": 0,
         "value_type": "individual"
      },
      "type": "graph",
      "xaxis": {
         "buckets": null,
         "mode": "time",
         "name": null,
         "show": true,
         "values": [ ]
      },
      "yaxes": [
         {
            "format": "short",
            "label": null,
            "logBase": 1,
            "max": null,
            "min": null,
            "show": true
         },
         {
            "format": "short",
            "label": null,
            "logBase": 1,
            "max": null,
            "min": null,
            "show": true
         }
      ]
   },
   "targets": {
      "aliasColors": { },
      "bars": false,
      "dashLength": 10,
      "dashes": false,
      "datasource": null,
      "fill": 1,
      "legend": {
         "avg": false,
         "current": false,
         "max": false,
         "min": false,
         "show": true,
         "total": false,
         "values": false
      },
      "lines": true,
      "linewidth": 1,
      "nullPointMode": "null",
      "percentage": false,
      "pointradius": 5,
      "points": false,
      "renderer": "flot",
      "repeat": null,
      "seriesOverrides": [ ],
      "spaceLength": 10,
      "span": 12,
      "stack": false,
      "steppedLine": false,
      "targets": [
         "foo",
         "foo"
      ],
      "thresholds": [ ],
      "timeFrom": null,
      "timeShift": null,
      "title": "with targets",
      "tooltip": {
         "shared": true,
         "sort": 0,
         "value_type": "individual"
      },
      "type": "graph",
      "xaxis": {
         "buckets": null,
         "mode": "time",
         "name": null,
         "show": true,
         "values": [ ]
      },
      "yaxes": [
         {
            "format": "short",
            "label": null,
            "logBase": 1,
            "max": null,
            "min": null,
            "show": true
         },
         {
            "format": "short",
            "label": null,
            "logBase": 1,
            "max": null,
            "min": null,
            "show": true
         }
      ]
   }
}
